pyqtgraph/graphicsItems/BoxplotItem.py

```diff
@@ -0,0 +1,288 @@
+        for pos, outliers in self.outlierData.items():
+            for o in outliers:
+                x, y = (pos, o) if self.opts["locAsX"] else (o, pos)
+                p.resetTransform()
+                p.translate(*tr.map(x, y))
+                p.scale(symbolSize, symbolSize)                
+                p.drawPath(symbol)
+                    
+    def boundingRect(self):
+        xmn, xmx = self.dataBounds(ax=0)
+        ymn, ymx = self.dataBounds(ax=1)
+        rect = QRectF(xmn, ymn, xmx-xmn, ymx-ymn)
+        
+        bpx = bpy = 0.0
+        spx = spy = 0.0
+        pxPad = self._pixelPadding
+        symbolPad = 0.7072 * (DEFAULT_SYM_SIZE if self.opts["symbolSize"] is None else self.opts["symbolSize"])
+        if pxPad > 0 or symbolPad > 0:
+            # determine length of pixel in local x, y directions
+            px, py = self.pixelVectors()
+            px = px.length() or 0
+            py = py.length() or 0
+            # return bounds expanded by pixel size
+            if pxPad > 0:
+                bpx = px * pxPad
+                bpy = py * pxPad
+            if symbolPad > 0:
+                spx = px * symbolPad
+                spy = py * symbolPad
+        
+        # bounding rect of boxes
+        rect = rect.adjusted(-bpx, -bpy, bpx, bpy)
+        # bounding rect of outliers
+        if self.opts["outlier"]:
+            if self.opts["locAsX"]:
+                rect = rect.adjusted(0, -spy, 0, spy)
+            else:
+                rect = rect.adjusted(-spx, 0, spx, 0)
+        return rect
+
+    def calculateDataBounds(self):
+        loc, data = self.opts["loc"], self.opts["data"]
+        if data is None:
+            return QRectF()
+
+        lst_lower = []
+        lst_upper = []
+        for dataset in data:
+            dataset = np.asarray(dataset)
+            if self.opts["outlier"]:
+                lower, upper = np.min(dataset), np.max(dataset)
+            else:
+                lower, upper = self.whiskerFunc(dataset)
+            lst_lower.append(lower)
+            lst_upper.append(upper)
+        miny = np.min(lst_lower)
+        maxy = np.max(lst_upper)
+
+        if loc is None:
+            loc = np.arange(len(data))
+        loc = np.array(loc)
+        minx, maxx = np.min(loc), np.max(loc)
+        width = self.opts["width"]
+        minx -= width/2
+        maxx += width/2
+
+        if not self.opts["locAsX"]:
+            minx, maxx, miny, maxy = miny, maxy, minx, maxx
+
+        return QRectF(QPointF(minx, miny), QPointF(maxx, maxy))
+
+    def dataBounds(self, ax, frac=1.0, orthoRange=None):
+        if self._dataBoundRect is None:
+            self._dataBoundRect = self.calculateDataBounds()
+            
+        if ax == 0:
+            return [self._dataBoundRect.left(), self._dataBoundRect.right()]
+        else:
+            return [self._dataBoundRect.top(), self._dataBoundRect.bottom()]
+
+    def pixelPadding(self):
+        return self._pixelPadding
```

Comment:
If you look at the implementation of ScatterPlotItem, symbol size padding also comes under pixelPadding()

Reply:
The outlier padding is a little bigger than expected, it is fixed in new commit.

Reply:
I am saying that pixelPadding() is supposed to report (half) the maximum of (symbolSize, penWidth).
pixelPadding() is queried by ViewBox.

boundingRect() should just call pixelPadding() to get the value.

Reply:
it should be easy to implement, but I do have few questions.

1. I see a value of sqrt(2) used as coefficient to get symbol padding, what does it mean? do I simply replace it with 0.5 as you suggested?
2. boundingRect() will call pixelPadding() to get pixel padding of QPen, the symbol padding is added only if outlier is enabled, if I use `(half) the maximum of (symbolSize, penWidth)`, the spaces around boxes might use symbol padding, is it expected?

Reply:
max( 0.7072 * symbolSize if show_outliers else 0, penWidth / 2)